kornia/geometry/epipolar/_metrics.py

```diff
@@ -12,45 +12,47 @@ def sampson_epipolar_distance(
 
     Args:
         pts1: correspondences from the left images with shape
-          (B, N, 2 or 3). If they are not homogeneous, converted automatically.
+          (*, N, 2 or 3). If they are not homogeneous, converted automatically.
         pts2: correspondences from the right images with shape
-          (B, N, 2 or 3). If they are not homogeneous, converted automatically.
-        Fm: Fundamental matrices with shape :math:`(B, 3, 3)`. Called Fm to
+          (*, N, 2 or 3). If they are not homogeneous, converted automatically.
+        Fm: Fundamental matrices with shape :math:`(*, 3, 3)`. Called Fm to
           avoid ambiguity with torch.nn.functional.
         squared: if True (default), the squared distance is returned.
         eps: Small constant for safe sqrt.
 
     Returns:
-        the computed Sampson distance with shape :math:`(B, N)`.
+        the computed Sampson distance with shape :math:`(*, N)`.
 
     """
     if not isinstance(Fm, torch.Tensor):
         raise TypeError(f"Fm type is not a torch.Tensor. Got {type(Fm)}")
 
-    if (len(Fm.shape) != 3) or not Fm.shape[-2:] == (3, 3):
+    if (len(Fm.shape) < 3) or not Fm.shape[-2:] == (3, 3):
```

Comment:
I'd suggest to use our new testing API: https://github.com/kornia/kornia/blob/master/kornia/testing/__init__.py#L185
`KORNIA_CHECK_SHAPE(Fm, ("*", "N", "3", "3")`

A similar thing could be added for checking the points